crates/starknet-handler/src/fact_registry/fact_registry.rs

```diff
@@ -1,58 +1,197 @@
-use primitive_types::H160;
+use proof_generator::model::account_proof::AccountProof;
 use starknet::{
-    accounts::{Account, Call, ExecutionEncoding, SingleOwnerAccount},
+    accounts::{
+        single_owner::SignError, Account, AccountError, Call, ExecutionEncoding, SingleOwnerAccount,
+    },
     core::{
-        chain_id,
-        types::{contract::SierraClass, FieldElement},
+        types::{BlockId, BlockTag, FieldElement, FunctionCall, InvokeTransactionResult},
         utils::get_selector_from_name,
     },
     providers::{
         jsonrpc::{HttpTransport, JsonRpcClient},
-        Url,
+        Provider, ProviderError, Url,
     },
     signers::LocalWallet,
 };
 
+use crate::util::get_high_and_low;
+
 pub struct FactRegistry {
     provider: JsonRpcClient<HttpTransport>,
     signer: LocalWallet,
-    _contract_artifact: SierraClass,
-    fact_registry: H160,
+    fact_registry: FieldElement,
+    owner_account: FieldElement,
 }
 
 #[allow(dead_code)]
 impl FactRegistry {
-    pub fn new(rpc: &str, fact_registry: H160, signer: LocalWallet) -> Self {
+    pub fn new(
+        rpc: &str,
+        fact_registry: FieldElement,
+        signer: LocalWallet,
+        owner_account: FieldElement,
+    ) -> Self {
         let url = Url::parse(rpc).unwrap();
         let provider = JsonRpcClient::new(HttpTransport::new(url));
 
-        let contract_artifact = include_bytes!("../artifacts/FactRegistry.json").to_vec();
-        let _contract_artifact = serde_json::from_slice::<SierraClass>(&contract_artifact).unwrap();
         Self {
             provider,
             signer,
-            _contract_artifact,
             fact_registry,
+            owner_account,
+        }
+    }
+
+    pub async fn prove_storage(
+        &self,
+        block_number: u64,
+        account_proof: AccountProof,
+        slot: String,
+    ) {
+        let (slot_high, slot_low) = get_high_and_low(slot.clone());
+        let bytes_len =
+            FieldElement::from_dec_str(account_proof.bytes.len().to_string().as_str()).unwrap();
```

Comment:
maybe better replacing unwrap() with proper error handling using Result and ? operator to avoid panics.

Reply:
custom errors added